src/meta/mod.rs

```diff
@@ -187,12 +187,18 @@ impl Meta {
     }
 
     pub fn from_path(path: &Path) -> Result<Self, std::io::Error> {
-        let metadata = if read_link(path).is_ok() {
+        let (metadata, symlink_is_dir) = if read_link(path).is_ok() {
             // If the file is a link, retrieve the metadata without following
-            // the link.
-            path.symlink_metadata()?
+            // the link, but provides whether followed link is_dir. Broken links are files.
+            (
+                path.symlink_metadata()?,
```

Comment:
This check would be much better done in `new` function.

This will avoid us passing `symlink_is_dir` thing around and the `expect('symlink should pass ...')`

Reply:
I suppose you mean `FileType::new`? I will refactor this.

Reply:
Yup.

Reply:
check the latest commit